integration_test.go

```diff
@@ -106,21 +127,82 @@ func setupTestRethEngine(t *testing.T) {
 	err = cli.ContainerStart(context.Background(), rethContainer.ID, container.StartOptions{})
 	require.NoError(t, err)
 
-	// TODO: check container status instead of sleeping
-	time.Sleep(5 * time.Second)
+	err = waitForRethContainer(t, jwtSecret)
+	require.NoError(t, err)
 
 	t.Cleanup(func() {
 		err = cli.ContainerStop(context.Background(), rethContainer.ID, container.StopOptions{})
 		require.NoError(t, err)
 		err = cli.ContainerRemove(context.Background(), rethContainer.ID, container.RemoveOptions{})
 		require.NoError(t, err)
-		err = os.Remove(DOCKER_JWTSECRET_PATH + DOCKER_JWT_SECRET_FILE)
+		err = os.Remove(jwtFile)
 		require.NoError(t, err)
 	})
+
+	return jwtSecret
+}
+
+// waitForRethContainer polls the reth endpoints until they're ready or timeout occurs
+func waitForRethContainer(t *testing.T, jwtSecret string) error {
+	t.Helper()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	client := &http.Client{
+		Timeout: 1 * time.Second,
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout waiting for reth container to be ready")
+		default:
+			// check :8545 is ready
+			rpcReq := strings.NewReader(`{"jsonrpc":"2.0","method":"net_version","params":[],"id":1}`)
+			resp, err := client.Post(TEST_ETH_URL, "application/json", rpcReq)
+			if err == nil {
+				resp.Body.Close()
+				if resp.StatusCode == http.StatusOK {
+					// check :8551 is ready
+					req, err := http.NewRequest("POST", TEST_ENGINE_URL, strings.NewReader(`{"jsonrpc":"2.0","method":"engine_exchangeTransitionConfigurationV1","params":[],"id":1}`))
+					if err != nil {
+						return err
+					}
+					req.Header.Set("Content-Type", "application/json")
+
+					jwtSecretBytes, err := hex.DecodeString(strings.TrimPrefix(jwtSecret, "0x"))
+					if err != nil {
+						return fmt.Errorf("failed to decode JWT secret: %w", err)
+					}
+
+					token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+						"iat": time.Now().Unix(),
+					})
+
+					signedToken, err := token.SignedString(jwtSecretBytes)
+					if err != nil {
+						return fmt.Errorf("failed to sign JWT token: %w", err)
+					}
+
+					req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", signedToken))
+
+					resp, err := client.Do(req)
+					if err == nil {
+						resp.Body.Close()
+						if resp.StatusCode == http.StatusOK {
+							return nil // Both endpoints are ready
+						}
+					}
+				}
+			}
+			time.Sleep(500 * time.Millisecond)
```

Comment:
the container is typically ready in 100 milliseconds in the local env. can reduce the poll interval. earlier, the tests failed with wait time of 50 milliseconds

Reply:
I agree that 100ms or even 50ms pooling time will be OK.